File: src/main/java/com/example/client_management/service/ClientService.java
package com.example.client_management.service;

import com.example.client_management.model.Client; 
import com.example.client_management.repository.ClientRepository; 
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ClientService {
    
    @Autowired
    private ClientRepository clientRepository; // Inyección del repositorio

    // Obtener todos los clientes
    public List<Client> getAllClients() {
        return clientRepository.findAll(); // Llama a findAll en el repositorio
    }

    // Obtener un cliente por su clave compartida
    public Optional<Client> getClientBySharedKey(String sharedKey) {
        return clientRepository.findBySharedKey(sharedKey); // Llama a findBySharedKey en el repositorio
    }

    // Agregar un nuevo cliente
    public void addClient(Client client) {
        clientRepository.save(client); // Guarda el cliente en el repositorio
    }
}

File: src/app/services/client.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Client } from '../models/client.model'; // Importa el modelo desde su archivo

const express = require('express');
const cors = require('cors');

const app = express();
app.use(cors());

@Injectable({
  providedIn: 'root'
})
export class ClientService {
  private apiUrl = 'http://localhost:4200/api/clients'; // Ajusta la URL según tu configuración

  constructor(private http: HttpClient) {}

  // Obtener todos los clientes
  getClients(): Observable<Client[]> {
    return this.http.get<Client[]>(this.apiUrl);
  }

  // Obtener un cliente por su clave compartida
  getClientBySharedKey(sharedKey: string): Observable<Client | null> {
    return this.http.get<Client | null>(`${this.apiUrl}?sharedKey=${sharedKey}`);
  }

  // Agregar un nuevo cliente
  addClient(client: Client): Observable<void> {
    return this.http.post<void>(this.apiUrl, client);
  }
}
